lca-chimevc-stack/lambda_functions/chime_call_analytics_initialization/index.js

```diff
@@ -626,12 +760,24 @@ const handler = async function handler(event, context) {
         );
         return;
       }
-      
-      callData = await getCallDataFromDdb(event.detail.callId);
-      await writeCallEndEventToKds(kinesisClient, callData.callId);
-      callData.callStreamingEndTime = new Date().toISOString();
-      await writeCallDataToDdb(callData, callData.callId);
-    } 
+
+      // VC streaming ENDED uses the original callId. Use originalCallId to get the callData.
+      const callDataFromDdb = await getCallDataWithOriginalCallIdFromDdb(event.detail.callId);
+      callDataFromDdb.callStreamingEndTime = new Date().toISOString();
+
+      await writeCallEndEventToKds(kinesisClient, callDataFromDdb.callId);
+      await writeCallDataToDdb(callDataFromDdb);
+
+      if (callDataFromDdb.recordingUrl) {
+        await writeS3UrlToKds(kinesisClient, callDataFromDdb.callId, callDataFromDdb.recordingUrl);
```

Comment:
Problem here is the recording still is not written..  In the old Call Transcriber Lambda, we do have a recording at the end of the call since it creates its own recording from the audio.. but with Chime Call Analytics the recording is created only by TCA, and comes in some time after the call is ended. So we need to call `writeS3UrlToKds` only when we process the TCA post call event.. i.e. from the `process_tca_postcall` lambda function.